llvm/lib/Target/AMDGPU/Utils/AMDGPUAsanInstrumentation.cpp

```diff
@@ -320,13 +320,11 @@ void getInterestingMemoryOperands(
     }
     default:
       for (unsigned ArgNo = 0; ArgNo < CI->arg_size(); ArgNo++) {
-        Value *Arg = CI->getArgOperand(ArgNo);
-        if (Arg->getType()->isPointerTy()) {
-          Type *Ty = Arg->getType();
+        if (Type *Ty = CI->getParamByRefType(ArgNo)) {
           Interesting.emplace_back(I, ArgNo, false, Ty, Align(1));
         } else if (CI->isByValArgument(ArgNo)) {
-          Type *Ty = CI->getParamByValType(ArgNo);
-          Interesting.emplace_back(I, ArgNo, false, Ty, Align(1));
+          Type *ValTy = CI->getParamByValType(ArgNo);
```

Comment:
The byval and byref APIs should behave the same. If you can directly query getParamByRefType for one you should be able to for the other